lib/server.js

```diff
@@ -10,8 +10,27 @@ module.exports = class HTTPServer extends TCPServer {
 
     super({ allowHalfOpen: false })
 
-    this.on('connection', (socket) => new HTTPServerConnection(this, socket, opts))
+    this._timeout = 0
+
+    this.on('connection', (socket) => {
+      const optsClone = JSON.parse(JSON.stringify(opts))
+      if (this._timeout) optsClone.timeout = this._timeout
```

Comment:
You can remove this and just use `this.server.timeout` in `HTTPServerConnection`.